src/Components/Aspire.OpenAI/AspireOpenAIClientBuilderChatClientExtensions.cs

```diff
@@ -0,0 +1,111 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Data.Common;
+using Aspire.OpenAI;
+using Microsoft.Extensions.AI;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using OpenAI;
+
+namespace Microsoft.Extensions.Hosting;
+
+/// <summary>
+/// Provides extension methods for registering <see cref="IChatClient"/> as a singleton in the services provided by the <see cref="IHostApplicationBuilder"/>.
+/// </summary>
+public static class AspireOpenAIClientBuilderChatClientExtensions
+{
+    private const string DeploymentKey = "Deployment";
+    private const string ModelKey = "Model";
+
+    /// <summary>
+    /// Registers a singleton <see cref="IChatClient"/> in the services provided by the <paramref name="builder"/>.
+    /// </summary>
+    /// <param name="builder">An <see cref="AspireOpenAIClientBuilder" />.</param>
+    /// <param name="deploymentName">Optionally specifies which model deployment to use. If not specified, a value will be taken from the connection string.</param>
+    /// <param name="disableOpenTelemetry">Optional. If <see langword="true"/>, skips registering open telemetry support in the <see cref="IChatClient"/> pipeline.</param>
+    /// <returns>A <see cref="ChatClientBuilder"/> that can be used to build a pipeline around the inner <see cref="IChatClient"/>.</returns>
+    public static ChatClientBuilder AddChatClient(
+        this AspireOpenAIClientBuilder builder,
+        string? deploymentName = null,
+        bool disableOpenTelemetry = false)
+    {
+        return builder.HostBuilder.Services.AddChatClient(
+            services => CreateInnerChatClient(services, builder, deploymentName, disableOpenTelemetry));
+    }
+
+    /// <summary>
+    /// Registers a keyed singleton <see cref="IChatClient"/> in the services provided by the <paramref name="builder"/>.
+    /// </summary>
+    /// <param name="builder">An <see cref="AspireOpenAIClientBuilder" />.</param>
+    /// <param name="serviceKey">The service key with which the <see cref="IChatClient"/> will be registered.</param>
+    /// <param name="deploymentName">Optionally specifies which model deployment to use. If not specified, a value will be taken from the connection string.</param>
+    /// <param name="disableOpenTelemetry">Optional. If <see langword="true"/>, skips registering open telemetry support in the <see cref="IChatClient"/> pipeline.</param>
+    /// <returns>A <see cref="ChatClientBuilder"/> that can be used to build a pipeline around the inner <see cref="IChatClient"/>.</returns>
+    public static ChatClientBuilder AddKeyedChatClient(
+        this AspireOpenAIClientBuilder builder,
+        string serviceKey,
+        string? deploymentName = null,
+        bool disableOpenTelemetry = false)
```

Comment:
I find the double-negatives a bit more challenging to reason about. Should this be enableOpenTelemetry = true?

Either way is ok, though.

Reply:
I was following the "optional flags should default to false" guidebook here. Even though the negation is clumsy, it makes it more obvious what the default is. A lot of ASP.NET Core flags work this way.

Totally willing to change this round if that's your preference. If you don't feel strongly I'll leave it as-is.

Reply:
> If you don't feel strongly I'll leave it as-is.

I don't.

Reply:
This is also what we decided for all our settings. Telemetry is enabled by default in Aspire components. Here is what we have:

```
    /// <summary>
    /// Gets or sets a boolean value that indicates whether the OpenTelemetry tracing is disabled or not.
    /// </summary>
    /// <value>
    /// The default value is <see langword="false"/>.
    /// </value>
    public bool DisableTracing { get; set; }
```

Reply:
No longer applicable as it now uses the higher-level config for this (instead of an explicit parameter).